core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,100 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * TODO: Insert description here. (generated by notcarl)
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
+    }
+
+    @Override
+    public OutputStream compress(OutputStream os) throws IOException {
+      return os;
+    }
+  };
+
+
+  /**
+   * Represents a message compressor.
+   */
+  public interface Compressor {
+    /**
+     * Returns the message encoding that this compressor uses.
+     *
+     * <p>This can be values such as "gzip", "deflate", "snappy", etc.
+     */
+    String getMessageEncoding();
```

Comment:
I had a hard time thinking of when you would want just a MessageEncoder string.  Did you have a use case in mind?